### PR TITLE
[Refactoring] Switch to @grpc/grpc-js, upgrade npm dependencies

### PR DESCRIPTION
**Pull request by pauliusuza**
### Refactoring to use @grpc/grpc-js

This pull request refactors the code base to use the official [@grpc/grpc-js](https://www.npmjs.com/package/@grpc/grpc-js) library written in Typescript, thus removing the need for native [grpc](https://www.npmjs.com/package/grp) dependency.

### Motivation 
I was not being able to install the etcd3 which used native grpc library on Windows 10. Where I could install it, native build added 2 minutes to production deployment time and configuration overhead.

Not only does this open up a door for this library to be used on the client side with some further work, it also reduces complexity and deployment time for production use cases.

### Other notable changes
- Upgraded Typescript to 3.8.3 syntax
- Updated most of the npm dependencies to the latest versions
- Updated `.proto` files
- Corrected the errors and syntax across the board to match the new changes
- Supports Node.js 14.0.0
- The `build:ts` command now works on windows (added shx library)
- Bumped library version to 0.3.0

All tests are green

### CODE REVIEW

**Comment by pauliusuza** [comment]
Any luck with getting maintainer access for @connor4312?
https://twitter.com/ConnorPeet/status/1253707039764647938

**Comment by tlhunter** [comment]
The _package.json_ version change might make it hard to merge (most module maintainers run the `npm version *` command after merging since it does other things like create git tags).

**Comment by connor4312** [comment]
Thanks for the PR!

I now have the necessary access -- we're now `microsoft/etcd3` instead of `mixer/etcd3`.

There are a bunch of changes in this PR and the Prettier update cause a lot of churn. Unfortunately I already did some housekeeping in master before starting to look at PRs/issues again, so there's a lot of conflicts. I'll look at your changes and apply them by hand into this repo.

File: proto/rpc.proto [hunk 1]
```diff
@@ -8,7 +8,7 @@ service KV {
   // Range gets the keys in the range from the key-value store.
```
Comment (connor4312) [inline proto/rpc.proto:8]: I will take these updates in a day or two; I want to update the proto files and add the appropriate new methods on our builders concurrently.

File: test/connection-pool.test.ts [hunk 2]
```diff
@@ -53,7 +53,9 @@ describe('connection pool', () => {
       .then(() => {
         throw new Error('expected to reject');
       })
-      .catch(err => expect(err).to.be.an.instanceof(GRPCConnectFailedError));
+      .catch((err) => {
+        expect(err).to.be.an.instanceof(GRPCGenericError);
```
Comment (connor4312) [inline test/connection-pool.test.ts:57]: The change here is actually updating this mapping:

https://github.com/microsoft/etcd3/blob/5c638a68674cb530d0ac81b2e3dcb0e8c52eb8bf/src/errors.ts#L208-L246

**Comment by connor4312** [comment]
I think I got all the changes in. Give me a shout if you notice anything missing.

I will publish 1.0.0 (I think 3 years is enough time to consider this 'stable') within the next few days.

**Comment by withSang** [comment]
I wonder if there's any update! :)

**Pull request closed without merging**